contracts/internal/EverlongERC4626.sol

```diff
@@ -1,39 +1,19 @@
 // SPDX-License-Identifier: Apache-2.0
-pragma solidity 0.8.20;
+pragma solidity 0.8.22;
 
+import { IHyperdrive } from "hyperdrive/contracts/src/interfaces/IHyperdrive.sol";
+import { HyperdriveUtils } from "hyperdrive/test/utils/HyperdriveUtils.sol";
+import { IERC20 } from "openzeppelin/interfaces/IERC20.sol";
 import { ERC4626 } from "solady/tokens/ERC4626.sol";
+import { EverlongPositions } from "./EverlongPositions.sol";
 
 /// @author DELV
 /// @title EverlongERC4626
 /// @notice Everlong ERC4626 implementation.
 /// @custom:disclaimer The language used in this code is for coding convenience
 ///                    only, and is not intended to, and does not, have any
 ///                    particular legal or regulatory significance.
-abstract contract EverlongERC4626 is ERC4626 {
-    // ╭─────────────────────────────────────────────────────────╮
-    // │ Storage                                                 │
-    // ╰─────────────────────────────────────────────────────────╯
-
-    /// @notice Virtual shares are used to mitigate inflation attacks.
-    bool public constant useVirtualShares = true;
-
-    /// @notice Used to reduce the feasibility of an inflation attack.
-    /// TODO: Determine the appropriate value for our case. Current value
-    ///       was picked arbitrarily.
-    uint8 public constant decimalsOffset = 3;
-
-    /// @dev Address of the token to use for Hyperdrive bond purchase/close.
-    address internal immutable _asset;
-
-    /// @dev Decimals used by the `_asset`.
-    uint8 internal immutable _decimals;
-
-    /// @dev Name of the Everlong token.
-    string internal _name;
-
-    /// @dev Symbol of the Everlong token.
-    string internal _symbol;
-
+abstract contract EverlongERC4626 is ERC4626, EverlongPositions {
```

Comment:
The way that we solved this kind of inheritance problem in Hyperdrive is to have the base contract contain an abstract function like `rebalance` (we did this with `checkpoint`). I prefer that style to the inheritance changes I'm seeing in this PR since the structure is flatter and easier to reason about.

Reply:
When I looked at Hyperdrive's `checkpoint` I didn't see any abstract functions used but maybe I'm missing it?

Regardless, I refactored to be more "flat" and is hopefully in line with what you meant. Currently running into the `Error (5005): Linearization of inheritance graph impossible` error I had mentioned before and pushed the repo in this state in case you want to take a look. 

The cause of this error is the addition of `IEverlong` to the list of extended items in `Everlong.sol`... seems very odd since none of the other contracts inherit `IEverlong`, they just reference errors from it.

Reply:
Seems like the hyperdrive contracts never do `Hyperdrive is IHyperdrive` for this reason, so removing the interface from the inheritance.